### PR TITLE
ENSIP: Wildcard writing/CCIP-Write

### PR DESCRIPTION
**Pull request by pikonha**
# ENSIP: CCIP-Write

## All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

## Description

Add ENSIP draft of the offchain writing standard.

## Changes
- [ ] New feature implementation
- [ ] Bug fix
- [ ] Code refactoring
- [X] Documentation update
- [ ] Other (please specify)

### CODE REVIEW

**Review by LeonardoVieira1630 — APPROVED** [review]
Captura de Tela 2024-09-26 às 08 42 17

Hi lucas. Great docs brother : )
The only thing that maybe we need to take a look is the images. For some reason i'm not able to se the architecture images. Is that right?

File: ensips/ccip-write/ccip-write.md [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+The function has the following signature:
+
+```solidity
+struct RegisterParams {
+    uint256 price;
+    bytes extraData;
+}
+
+function registerParams(
+    bytes memory name,
```
Comment (alextnetto) [inline ensips/ccip-write/ccip-write.md:57]:
```suggestion
    bytes memory name,
    uint256 commitTime,
```

If commitTime == 0 then there is no commit reveal. If there is commit then it's the keccak256(registration calldata)

File: ensips/ccip-write/ccip-write.md [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+
+- `price`: the amount of ETH charger per second
+- `extraData`: any given structure in an ABI encoded format
+
+Sample of callback function used to validate and handle the response returned by the L2 contract:
+
+```solidity
+function registerParamsCallback(
+    bytes[] memory values,
+    bytes memory extraData
```
Comment (alextnetto) [inline ensips/ccip-write/ccip-write.md:80]: commitTime here too

File: ensips/ccip-write/ccip-write.md [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+```solidity
+function register(
+    string calldata name,
+    address owner,
+    uint256 duration,
+    bytes32 secret,
+    address resolver,
+    bytes[] calldata data,
+    bool reverseRecord,
+    uint16 fuses
```
Comment (alextnetto) [inline ensips/ccip-write/ccip-write.md:103]:
```suggestion
    uint16 fuses,
    bytes memory extraData
```

Important for being able to have a more flexible logic or usecases on the registration

File: ensips/ccip-write/ccip-write.md [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+    uint256 duration
+)
+    external
+    view
+    returns (RegisterParams memory);
+```
+
+Parameters:
+
+- `name`: The DNS-encoded name to query
```
Comment (alextnetto) [inline ensips/ccip-write/ccip-write.md:67]:
```suggestion
- `name`: The DNS-encoded name to be registered
```

File: ensips/ccip-write/ccip-write.md [hunk 5]
```diff
@@ -0,0 +1,342 @@
...
+
+The interface for enabling domain transfers MUST be implemented by both the resolver deployed to Ethereum and the contract responsible for managing domains deployed on the layer 2. The implementation differs between L1 and L2 as follows:
+
+1. **L1 Resolver**: MUST revert with the respective error described by EIP-5559, indicating that the actual transfer should occur elsewhere.
+2. **L2 Contract**: MUST handle the actual domain transfer operation.
+
+The transfer function MUST have the following signature:
+
+```solidity
+function transfer(bytes32 node, address owner) external;
```
Comment (alextnetto) [inline ensips/ccip-write/ccip-write.md:133]:
```suggestion
function transfer(bytes32 node, address to) external;
```

File: ensips/ccip-write/ccip-write.md [hunk 6]
```diff
@@ -0,0 +1,342 @@
...
+```
+
+With the arguments being:
+
+1. `node`: a valid ENS node (namehash).
+2. `owner`: an Ethereum address.
```
Comment (alextnetto) [inline ensips/ccip-write/ccip-write.md:139]:
```suggestion
2. `to`: the Ethereum address to receive the domain.
```

File: ensips/ccip-write/ccip-write.md [hunk 7]
```diff
@@ -0,0 +1,342 @@
...
+
+```solidity
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+/**
+ * @title IOffchainResolver Interface
+ * @dev Interface for a resolver that supports text and address records management, with additional functions for registration, ownership transfer, and multicall.
+ */
+interface IOffchainResolver {
```
Comment (alextnetto) [inline ensips/ccip-write/ccip-write.md:283]: Apply changes from above here. 

Also add the commit interface and the explanation for it

**Review by alextnetto — COMMENTED** [review]
great job!!

**Pull request merged**